src/main/java/co/cask/PluginJSON.java

```diff
@@ -256,6 +333,14 @@ private void initialize() {
       widgetDirectory = new File(widgetsDirectory);
     }
 
+    if (iconsDirectory != null) {
+      if (!iconsDirectory.contains("/")) {
+        iconDirectory = new File(project.getBasedir() + "/" + iconsDirectory);
```

Comment:
this entire block of code is repeated 3 times, please move it to a separate method so it's done the same way and to fix the existing bugs for widgets and docs directories